test/functional/options/num_options_spec.lua

```diff
@@ -100,19 +101,20 @@ describe(':set validation', function()
   it('set wmh/wh wmw/wiw checks', function()
     feed_command('set winheight=2')
     feed_command('set winminheight=3')
-    eq('E591', eval('v:errmsg'):match('E%d*'))
+    matches('E591', eval('v:errmsg'))
 
     feed_command('set winwidth=2')
     feed_command('set winminwidth=3')
-    eq('E592', eval('v:errmsg'):match('E%d*'))
+    matches('E592', eval('v:errmsg'))
   end)
 
   it('set maxcombine resets to 6', function()
     local function setto(value)
       feed_command('setglobal maxcombine=' .. value)
       feed_command('setlocal maxcombine=' .. value)
-      api.nvim_set_option_value('maxcombine', value, {})
-      eq(6, api.nvim_get_option_value('maxcombine', {}))
+      feed_command('set maxcombine=' .. value)
+      eq(6, eval('&maxcombine'))
+      matches('maxcombine=6', exec_capture('set maxcombine?'))
```

Comment:
why use matches for this instead of eq() with the actual option value? this would incorrectly pass if e.g. `maxcombine=666` was returned